Switch uses of __import__ to importlib.get_module()

jdufresne opened the pull request:
The Python docs for `__import__` recommend using `importlib.get_module()`.

https://docs.python.org/3/library/functions.html#__import__

> Note: This is an advanced function that is not needed in everyday Python programming, unlike `importlib.import_module()`.

As `importlib.get_module()` uses the Python module cache and returns the module, this also allows simplifying many module cache checks of use of `sys.modules`.

`importlib.get_module()` has been available since Python 3.3.

tk0miya reviewed (changes requested):
LGTM. Could you rebase this into 2.0 branch?

jdufresne committed:
Switch uses of __import__ to importlib.get_module()

The Python docs for __import__ recommend using importlib.get_module().

https://docs.python.org/3/library/functions.html#__import__

> Note: This is an advanced function that is not needed in everyday
> Python programming, unlike importlib.import_module().

As importlib.get_module() uses the Python module cache and returns the
module, this also allows simplifying many module cache checks of use of
sys.modules.

importlib.get_module() has been available since Python 3.3.

jdufresne commented:
Sure thing. Rebased.

tk0miya reviewed (approved):
Thank you for quick work!
